python/lsst/meas/algorithms/starSelector.py

```diff
@@ -0,0 +1,154 @@
+            for star in psfRes.goodStarCat:
+                star.set(isStarKey, True)
+
+        return pipeBase.Struct(
+            starCat = psfRes.goodStarCat,
+            psfCandidates = psfRes.psfCandidates,
+        )
+
+    @abc.abstractmethod
+    def selectStars(self, exposure, sourceCat, matches=None):
+        """!Return a catalog of stars: a subset of sourceCat
+        
+        A list of PSF candidates may be used by a PSF fitter to construct a PSF.
+        
+        @param[in] exposure  the exposure containing the sources
+        @param[in] sourceCat  catalog of sources that may be stars (an lsst.afw.table.SourceCatalog)
+        @param[in] matches  astrometric matches; ignored by this star selector
+                (an lsst.afw.table.ReferenceMatchVector), or None. Some star selectors
+                will ignore this argument, others may require it. See the usesMatches class variable.
+        
+        @return a pipeBase.Struct containing:
+        - starCat  a catalog of stars
+        """
+        raise NotImplementedError("StarSelectorTask is abstract, subclasses must override this method")
+
+    def makePsfCandidates(self, exposure, starCat):
+        """!Make a list of PSF candidates from a star catalog
+
+        @param[in] exposure  the exposure containing the sources
+        @param[in] starCat  catalog of stars (an lsst.afw.table.SourceCatalog),
+                            e.g. as returned by the run or selectStars method
+
+        @return an lsst.pipe.base.Struct with fields:
+        - psfCandidates  list of PSF candidates (lsst.meas.algorithms.PsfCandidate)
+        - goodStarCat  catalog of stars that were successfully made into PSF candidates (a subset of starCat)
+        """
+        goodStarCat = SourceCatalog(starCat.schema)
+
+        psfCandidateList = []
+        for star in starCat:
+            try:
+                psfCandidate = algorithmsLib.makePsfCandidate(star, exposure)
+                
+                # The setXXX methods are class static, but it's convenient to call them on
+                # an instance as we don't know Exposure's pixel type
+                # (and hence psfCandidate's exact type)
+                if psfCandidate.getWidth() == 0:
+                    psfCandidate.setBorderWidth(self.config.borderWidth)
+                    psfCandidate.setWidth(self.config.kernelSize + 2*self.config.borderWidth)
+                    psfCandidate.setHeight(self.config.kernelSize + 2*self.config.borderWidth)
+
+                im = psfCandidate.getMaskedImage().getImage()
+                vmax = afwMath.makeStatistics(im, afwMath.MAX).getValue()
+                if not np.isfinite(vmax):
+                    continue
+                psfCandidateList.append(psfCandidate)
+                goodStarCat.append(star)
+            except Exception as err:
+                self.log.warn("Failed to make a psfCandidate from star %d: %s" % (star.getId(), err))
```

Comment:
It might be better to just wrap the call to `makePsfCandidate` in a try block, as I don't think we expect it to be possible that the rest could throw (and hence we'd want the exception to propagate up and cause an early failure if it did, so we could fix the bug).